src/daft-core/src/series/ops/agg.rs

```diff
@@ -276,4 +276,78 @@ impl Series {
             ))),
         }
     }
+
+    pub fn bool_and(&self, groups: Option<&GroupIndices>) -> DaftResult<Self> {
+        use crate::array::ops::DaftBoolAggable;
+        match self.data_type() {
+            DataType::Boolean => {
+                let downcasted = self.bool()?;
+                match groups {
+                    Some(groups) => {
+                        Ok(DaftBoolAggable::grouped_bool_and(downcasted, groups)?.into_series())
+                    }
+                    None => {
+                        if self.is_empty() {
+                            Ok(Self::full_null(
+                                self.field().name.as_str(),
+                                &DataType::Boolean,
+                                1,
+                            ))
+                        } else {
+                            Ok(DaftBoolAggable::bool_and(downcasted)?.into_series())
+                        }
```

Comment:
Hmm, maybe instead of performing the `if self.empyt() { .. } else { .. }` check inside of `Series::bool_and`, the check should be performed inside of `ListArray::bool_and` and `FixedSizeListArray::bool_and`?

Reply:
Suggestion:

```rs
let downcasted = self.bool()?;
Ok(match groups {
  Some(groups) => self.grouped_bool_and(groups)?.into_series(),
  None => self.bool_and()?.into_series(),
})
```

And inside of `Bool::bool_and` and is where we perform the is-empty check.

My reasoning is that the is-empty check should be performed at the core-most API level. Say in the future, a function named `foo` wants to call `Bool::bool_and`, `foo` doesn't need to perform the is-empty check itself.